frontend/views/containers/GroupPledgesGraph.vue

```diff
@@ -1,218 +1,195 @@
 
 export default {
   name: 'GroupPledgesGraph',
   components: {
     PieChart,
-    GraphLegendGroup,
     GraphLegendItem,
     Tooltip
   },
   props: {
-    // Needed to have a dynamic graph (ex: used on Contributions page)
-    userPledgeAmount: {
-      type: Number,
+    type: {
+      type: String, // incomeAmount || pledgeAmount TODO validator
       default: null
     },
-    userIncomeAmount: {
+    amount: {
       type: Number,
       default: null
     }
   },
-  data () {
-    return {
-      // REVIEW - make this a watcher
-      isMobile: this.verifyIsMobile(),
-      // -- Hardcoded Data just for layout purpose:
-      fakeStore: {
-        currency: currencies.USD.symbol,
-        groupMembersTotal: 7,
-        groupPledgeGoal: 800,
-        mincome: 500,
-        othersPledges: [180, 130, 200]
-      }
-    }
-  },
-  mounted: function () {
-    window.addEventListener('resize', this.handleResize)
-  },
-  beforeDestroy: function () {
-    window.removeEventListener('resize', this.handleResize)
-  },
   computed: {
-    chartSize () {
-      return this.isMobile ? '8rem' : undefined
-    },
-    legendItemClass () {
-      return 'column c-legend-item'
-    },
+    ...mapGetters([
+      'groupSettings',
+      'groupProfiles',
+      'groupMembersCount',
+      'ourContractId'
+    ]),
     graphData () {
-      const { userPledgeAmount, userIncomeAmount } = this
-      const { groupMembersTotal, groupPledgeGoal, mincome, othersPledges } = this.fakeStore
-      const othersAmount = othersPledges.reduce((acc, cur) => acc + cur, 0)
-      const userIncomeNeeded = userIncomeAmount !== null && userIncomeAmount < mincome ? mincome - userIncomeAmount : null
-      const totalAmount = othersAmount + userPledgeAmount
-      const goal = groupPledgeGoal + userIncomeNeeded
-      const members = othersPledges.length + (userPledgeAmount ? 1 : 0)
-      const neededPledges = goal - totalAmount
-      const surplus = totalAmount - groupPledgeGoal
-      const membersNeedingPledges = groupMembersTotal - othersPledges.length
-
-      // Dumb Rule of 3 algorithm to know how much the user will receive (userIncomeToReceive) - just for layout purposes.
-      const avgMembersNeedingPledges = goal / membersNeedingPledges
-      const avgNeededPledges = neededPledges / membersNeedingPledges
-      const userIncomeToReceive = userIncomeNeeded ? (userIncomeNeeded - (userIncomeNeeded * avgNeededPledges / avgMembersNeedingPledges)).toFixed(2) : 0
+      const mincome = this.groupSettings.mincomeAmount
+      const ourPledgeAmount = this.type === 'pledgeAmount' && this.amount >= 0 && this.amount
+      const ourIncomeAmount = this.type === 'incomeAmount' && this.amount < mincome && this.amount
+      const doWeNeedPledge = typeof ourIncomeAmount === 'number'
+      let othersIncomeNeeded = 0
+      let othersPledgesAmount = 0
+      let othersPledgesCount = 0
+
+      Object.keys(this.groupProfiles).forEach(username => {
```

Comment:
for/in loops are better in this sort of situation since they're faster. Line 85 will need to be updated if you convert it